src/compiler/binder.ts

```diff
@@ -2310,27 +2310,20 @@ namespace ts {
             if (node.modifiers && node.modifiers.length) {
                 file.bindDiagnostics.push(createDiagnosticForNode(node, Diagnostics.Modifiers_cannot_appear_here));
             }
-
-            if (node.parent.kind !== SyntaxKind.SourceFile) {
-                file.bindDiagnostics.push(createDiagnosticForNode(node, Diagnostics.Global_module_exports_may_only_appear_at_top_level));
-                return;
+            const diag = !isSourceFile(node.parent)
+                ? Diagnostics.Global_module_exports_may_only_appear_at_top_level

```

Comment:
Since this is a cleanup, can you instead use the syntax

```ts
preducate1 ? result1 :
predicate2 ? result2 :
undefined
```

I think that's a lot easier to read.